src/core/definePage.ts

```diff
@@ -84,6 +103,16 @@ export function definePageTransform({
 
     s.remove(setupOffset + routeRecord.end!, code.length)
     s.remove(0, setupOffset + routeRecord.start!)
+    if (scriptSetup.lang === 'ts') {
+      const transformedResults = transformSync(
+        `export default ${s.toString()}`,
+        { plugins: ['@babel/plugin-transform-typescript'] }
+      )?.code
```

Comment:
Did you benchmark this with vite plugin inspect? I think it's probably too slow 🤔

Reply:
I haven't benchmarked it with vite plugin inspect. First time modifying vite plugin. I'll check it a bit later.